lib/path.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const {PATH_SEPARATOR} = require('./constants');
+const isArray = require('./is-array');
+const isSymbol = require('./is-symbol');
+
+module.exports = {
+	after: (path, subPath) => {
+		if (isArray(path)) {
+			return path.slice(subPath.length);
+		}
+
+		if (subPath === '') {
+			return path;
+		}
+
+		return path.slice(subPath.length + 1);
+	},
+	concat: (path, key) => {
+		if (isArray(path)) {
+			path = path.slice();
+
+			if (key) {
+				path.push(key);
+			}
+
+			return path;
+		}
+
+		if (key && key.toString !== undefined) {
+			if (path !== '') {
+				path += PATH_SEPARATOR;
+			}
+
+			if (isSymbol(key)) {
+				return path + 'Symbol(' + key.description + ')';
+			}
+
+			return path + key;
+		}
+
+		return path;
+	},
+	initial: path => {
+		if (isArray(path)) {
+			return path.slice(0, -1);
+		}
+
+		if (path === '') {
+			return path;
+		}
+
+		const index = path.lastIndexOf(PATH_SEPARATOR);
+
+		if (index === -1) {
+			return '';
+		}
+
+		return path.slice(0, index);
+	},
+	walk: (path, callback) => {
+		if (isArray(path)) {
+			path.forEach(callback);
+		} else if (path !== '') {
+			let pos = 0;
```

Comment:
```suggestion
			let position = 0;
```

More readable to not use abbreviations.